### PR TITLE
CHK-1518: Query pagination

### PR DESCRIPTION
**Pull request by pietro-tota**
Ecommerce-local related [PR](https://github.com/pagopa/pagopa-ecommerce-local/pull/74) (add env properties)

#### List of Changes

The aim of this PR is to configure TPS (transactions per seconds) rate at which pending transactions are processed by batch.

This has been done introducing query pagination, so transactions are analyzed by chunk were:

- `PENDING_TRANSACTIONS_MAX_TRANSACTIONS_PER_PAGE` parameter allow to specify how many transactions to retrieve for each page -> chunk size
- `PENDING_TRANSACTIONS_PAGE_ANALYSIS_DELAY_SECONDS` parameter allow to specify time to be waited between the next page to be analyzed -> chunk rate

Changing the two above parameters value can change the maximum configured for pending transaction analysis.

Example:

| Max transaction per page | Delay between pages | Max TPS |
|--------------------------|---------------------|---------|
| 1                        | 1                   | 1       |
| 5                        | 1                   | 5       |
| 5                        | 2                   | 2,5     |

#### Motivation and Context

As for the previous implementations all pending transactions were retrieved used Mongo Reactive Crud repository for retrieve all pending transactions (returning a Flux<Transaction> from which pending transaction processing start for each of the found transactions).
This approach is the fastest since pending transactions are scanned through a selector so all transactions are processed one-by-one (see https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#repositories.scrolling.guidance)

The problem with this approach is that in case of a large number of pending transaction this will result in a constant increase in consumed RU and CPU resources that can lead to problems because of the processing resulting throttling.

With this change the TPS is known a priori and can be tuned to balance performance and consumed resources making batch executions predictable.

#### How Has This Been Tested?

Junit tests + Local test with ecommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -12,4 +12,6 @@ pendingTransactions.batch.scheduledChron=${PENDING_TRANSACTIONS_SCHEDULE_CRON}
 pendingTransactions.batch.transactionsAnalyzer.executionRateMultiplier=${PENDING_TRANSACTIONS_WINDOWS_BATCH_EXECUTION_RATE_MULTIPLIER}
 pendingTransactions.batch.transactionsAnalyzer.parallelEventsToProcess=${PENDING_TRANSACTIONS_PARALLEL_EVENTS_TO_PROCESS}
 pendingTransactions.batch.maxDurationSeconds=${PENDING_TRANSACTIONS_MAX_DURATION_SECONDS:-1}
-pendingTransactions.batch.sendPaymentResultTimeout=${PENDING_TRANSACTIONS_SEND_PAYMENT_RESULT_TIMEOUT_SECONDS}
\ No newline at end of file
+pendingTransactions.batch.maxTransactionsPerPage=${PENDING_TRANSACTIONS_MAX_TRANSACTIONS_PER_PAGE}
```
Comment (Fernando-Granato) [inline src/main/resources/application.properties:15]: maxTransactionsPerPage 👉🏻 maxTransactions**For**Page what do you think?

File: helm/values-dev.yaml [hunk 2]
```diff
@@ -78,11 +78,13 @@ microservice-chart:
     PENDING_TRANSACTIONS_SCHEDULE_CRON: "0 0 */1 * * *"
     PENDING_TRANSACTIONS_WINDOWS_BATCH_EXECUTION_RATE_MULTIPLIER: "2"
     PENDING_TRANSACTIONS_PARALLEL_EVENTS_TO_PROCESS: "1"
+    PENDING_TRANSACTIONS_MAX_TRANSACTIONS_PER_PAGE: "5"
+    PENDING_TRANSACTIONS_PAGE_ANALYSIS_DELAY_SECONDS: "1"
```
Comment (Fernando-Granato) [inline helm/values-dev.yaml:82]: Could it be just a delay of 1s for the execution of a page?

File: src/main/resources/application.properties (hunk 1 above)
Comment (pietro-tota) [reply src/main/resources/application.properties:15]: `per` here is used for indicate a rate, so I would leave it as for now

File: helm/values-dev.yaml (hunk 2 above)
Comment (pietro-tota) [reply helm/values-dev.yaml:82]: Those parameters can be changed any time with the proper values.
As an initial value I valued those parameters for have max 5 transaction per seconds processing rate that is a reasonable value considering that event dispatcher process 10 event per seconds.

Then those values can be adjusted after some time in order to put the wanted TPS as per the environment resource limits such as CosmosDB RU etc.

**Pull request merged**